### PR TITLE
Adds presubmit job for cert-manager/istio-csr targeting istio v1.11.4

### PR DESCRIPTION
**Pull request by JoshVanL**
Signed-off-by: joshvanl <vleeuwenjoshua@gmail.com>

Also updates all the images in the file to use Go v1.17

/assign @irbekrm 
/hold

### CODE REVIEW

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml [hunk 1]
```diff
@@ -282,3 +282,57 @@ presubmits:
...
+    agent: kubernetes
+    decorate: true
+    branches:
+    - ^main$
+    labels:
+      preset-service-account: "true"
+      preset-dind-enabled: "true"
+    spec:
+      containers:
+      - image: eu.gcr.io/jetstack-build-infra-images/golang-dind:20210331-a8721c1-1.16
```
Comment (irbekrm) [inline config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:302]: There is a Go 1.17 image now https://console.cloud.google.com/gcr/images/jetstack-build-infra-images/eu/golang-dind?authuser=2&project=jetstack-build-infra-images&tab=vulnz

Comment (JoshVanL) [reply config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml]: Thanks, I went ahead and updated all the images in the file.

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml [hunk 2]
```diff
@@ -282,3 +282,57 @@ presubmits:
         options:
           - name: ndots
             value: "1"
+
+  # kind based istio-csr e2e job for Kubernetes v1.21, istio v1.11
+  - name: pull-cert-manager-istio-csr-k8s-v1-21-istio-v1-11
+    context: pull-cert-manager-istio-csr-k8s-v1-21-istio-v1-11
+    # Match everything except PRs that only touch docs/
```
Comment (irbekrm) [inline config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:289]: I think we don't actually skip the docs changes in this job? (should be a `skip_if_only_changed` field https://github.com/kubernetes/test-infra/blob/master/prow/jobs.md#triggering-jobs-based-on-changes)
Same with the other presubmit

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml [hunk 3]
```diff
@@ -282,3 +282,57 @@ presubmits:
...
+        - runner
+        - make
+        - e2e
+        resources:
+          requests:
+            cpu: 3500m
+            memory: 12Gi
+        env:
+        - name: K8S_VERSION
+          value: "1.21.1"
```
Comment (irbekrm) [inline config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:313]: Should it maybe be the latest kube (1.22)?

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml (hunk 2 above)
Comment (JoshVanL) [reply config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:289]: Thanks, I think I want to just always run these jobs for all PRs. I will remove the false comments.

File: config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml (hunk 3 above)
Comment (JoshVanL) [reply config/jobs/cert-manager/istio-csr/cert-manager-istio-csr-presubmits.yaml:308]: Yep, sounds good to me :+1:

**Comment by irbekrm** [comment]
Thanks for all the changes

/lgtm
/approve

**Comment by JoshVanL** [comment]
@irbekrm needs a GH approval

**Review by irbekrm — APPROVED** [review]

**Pull request merged**